lib/windows_error/h_result/facility.rb

```diff
@@ -0,0 +1,200 @@
+module WindowsError
+  module HResult
+    module Facility
+      FacilityCode = Struct.new('FacilityCode', :name, :value, :description) do
+        def ==(other_object)
+          if other_object.kind_of? self.class
+            self.value == other_object.value
+          elsif other_object.kind_of? Integer
+            self.value == other_object
+          elsif other_object.nil?
+            false
+          else
+            raise ArgumentError, "Cannot compare a #{self.class} to a #{other_object.class}"
+          end
+        end
+
+        alias :=== :==
+
+        def to_s
+          code = sprintf "%04x", self.value
+          "(0x#{code}) #{self.name}: #{self.description}"
+        end
+      end
+
+      # Return the {WindowsError::HResult::FacilityCode} object that matches
+      # the value supplied.
+      #
+      # @param [Integer] retval the return value you want the facility code for
+      # @raise [ArgumentError] if something other than a Integer is supplied
+      # @return [WindowsError::HResult::FacilityCode, Nil] the FacilityCode that matched
+      def self.find_by_code(retval)
```

Comment:
Unline some of the other error codes that can have duplicate values, there are no duplicate Facility codes so this only returns one value instead of an array value. Since the return type is a single object instead of an array, it's not called `find_by_retval`. Also the facility code isn't a return value so that wouldn't really make sense.